crates/runtime/src/capability/builtin.rs

```diff
@@ -188,94 +188,73 @@ impl Eq for ActorIdentifier {}
+        let (wit_ns, wit_pkg, wit_iface, _) = self.interface.as_parts();
+        let link_name = &self.link_name;
+        let id = &self.id;
+        write!(f, "{link_name}/{id}/{wit_ns}:{wit_pkg}/{wit_iface}")
+    }
+}
+
 #[derive(Clone, Debug, Eq, PartialEq)]
 /// Target entity
 pub enum TargetEntity {
     /// Link target entity
+    #[deprecated(note = "use the wrpc variant instead")]
     Link(Option<String>),
     /// Actor target entity
+    #[deprecated(note = "use the wrpc variant instead")]
     Actor(ActorIdentifier),
     /// WRPC component
-    Wrpc(InterfaceTarget),
-}
-
-impl TryFrom<bus::lattice::ActorIdentifier> for ActorIdentifier {
-    type Error = anyhow::Error;
-
-    fn try_from(entity: bus::lattice::ActorIdentifier) -> Result<Self, Self::Error> {
-        match entity {
-            bus::lattice::ActorIdentifier::PublicKey(key) => {
-                let key =
-                    KeyPair::from_public_key(&key).context("failed to parse actor public key")?;
-                Ok(ActorIdentifier::Key(Arc::new(key)))
-            }
-            bus::lattice::ActorIdentifier::Alias(alias) => Ok(ActorIdentifier::Alias(alias)),
-        }
-    }
-}
-
-impl TryFrom<bus::lattice::TargetEntity> for TargetEntity {
-    type Error = anyhow::Error;
-
-    fn try_from(entity: bus::lattice::TargetEntity) -> Result<Self, Self::Error> {
-        match entity {
-            bus::lattice::TargetEntity::Link(name) => Ok(Self::Link(name)),
-            bus::lattice::TargetEntity::Actor(actor) => actor.try_into().map(TargetEntity::Actor),
-        }
-    }
+    Wrpc(WrpcInterfaceTarget),
 }
 
 #[derive(Debug, Clone, Eq, Hash, PartialEq)]
-/// Call target identifier
-pub enum TargetInterface {
-    /// `wasi:blobstore/blobstore`
-    WasiBlobstoreBlobstore,
-    /// `wasi:http/outgoing-handler`
-    WasiHttpOutgoingHandler,
-    /// `wasi:keyvalue/atomic`
-    WasiKeyvalueAtomic,
-    /// `wasi:keyvalue/eventual`
-    WasiKeyvalueEventual,
-    /// `wasi:logging/logging`
-    WasiLoggingLogging,
-    /// `wasmcloud:messaging/consumer`
-    WasmcloudMessagingConsumer,
-    /// Custom interface
-    Custom {
-        /// Package namespace
-        namespace: String,
-        /// Package name
-        package: String,
-        /// Interface name
-        interface: String,
-    },
-}
-
-impl TargetInterface {
+/// Call target identifier, which is equivalent to a WIT specification, which
+/// can identify an interface being called and optionally a specific function on that interface.
+pub struct CallTargetInterface {
+    /// WIT namespace (ex. `wasi` in `wasi:keyvalue/readwrite.get`)
+    pub namespace: String,
+    /// WIT package name (ex. `keyvalue` in `wasi:keyvalue/readwrite.get`)
+    pub package: String,
+    /// WIT interface (ex. `readwrite` in `wasi:keyvalue/readwrite.get`)
+    pub interface: String,
+    /// WIT package name (ex. `readwrite` in `wasi:keyvalue/readwrite.get`)
+    pub function: Option<String>,
```

Comment:
nit: I think this should be `get` right?

We definitely need to rethink the target interface things. I might change some of this up in #1568 depending on what I can see is obsolete

Reply:
Yup, will fix!